ansible/inventory/multi_inventory.py

```diff
@@ -5,29 +5,294 @@
+        for to_name, from_name in self.config.get('clone_groups', {}).items():
+            for name, data in self.inventory['_meta']['hostvars'].items():
+                if 'synthetic_' in name:
+                    continue
+
+                key = '%s_%s' % (to_name, MultiInventoryUtils.get_entry(data, from_name))
+                if not self.inventory.has_key(key):
+                    self.inventory[key] = []
+                self.inventory[key].append(name)
+
+    def apply_group_selectors(self):
+        """Apply the account config for group selectors """
+        _ = self # Here for pylint.  wanted an instance method instead of static
+        # There could be multiple clusters per account.  We need to process these selectors
+        # based upon the oo_clusterid_ variable.
+        clusterids = [group for group in self.inventory if "oo_clusterid_" in group]
+
+        for clusterid in clusterids:
+            for selector in self.config.get('group_selectors', {}):
+                if self.inventory.has_key(selector['from_group']):
+                    hosts = list(set(self.inventory[clusterid]) & set(self.inventory[selector['from_group']]))
+                    hosts.sort()
+
+                    # Multiple clusters in an account
+                    if self.inventory.has_key(selector['name']):
+                        self.inventory[selector['name']].extend(hosts[0:selector['count']])
+                    else:
+                        self.inventory[selector['name']] = hosts[0:selector['count']]
+
+                    for host in hosts:
+                        if host in self.inventory[selector['name']]:
+                            self.inventory['_meta']['hostvars'][host][selector['name']] = True
+                        else:
+                            self.inventory['_meta']['hostvars'][host][selector['name']] = False
+
+
+    def apply_account_config(self):
+        """ Apply account config settings """
+        self.inventory['all_hosts'] = self.inventory['_meta']['hostvars'].keys()
+
+        self.apply_cluster_vars()
+
+        self.apply_extra_vars()
+
+        self.apply_clone_vars()
+
+        self.apply_extra_groups()
+
+        self.apply_clone_groups()
+
+        self.apply_group_selectors()
+
+        # store the results back into all_inventory_results
+        #return self.inventory
+
+    def generate_config(self):
+        """Generate the provider_files in a temporary directory"""
+        prefix = 'multi_inventory.'
+        tmp_dir_path = tempfile.mkdtemp(prefix=prefix)
+        atexit.register(MultiInventoryUtils.cleanup, [tmp_dir_path])
+        for provider_file in self.config['provider_files']:
+            with open(os.path.join(tmp_dir_path, provider_file['name']), 'w+') as filedes:
+                content = Template(provider_file['contents']).substitute(tmpdir=tmp_dir_path)
+                filedes.write(content)
+
+        return tmp_dir_path
+
+    def write_to_cache(self):
```

Comment:
Consider adding a parameter for what to write (and remove calling self.inventory).